pkg/operator/certrotation/cabundle_test.go

```diff
@@ -64,6 +63,66 @@ func TestEnsureConfigMapCABundle(t *testing.T) {
 				if len(actual.Data["ca-bundle.crt"]) == 0 {
 					t.Error(actual.Data)
 				}
+				if len(actual.OwnerReferences) != 1 {
+					t.Errorf("expected to have exactly one owner reference")
+				}
+				if actual.OwnerReferences[0].Name != "operator" {
+					t.Errorf("expected owner reference to be 'operator', got %v", actual.OwnerReferences[0].Name)
+				}
+				if got, exists := actual.Annotations["openshift.io/owning-component"]; !exists || got != "test" {
+					t.Errorf("owner annotation is missing: %#v", actual.Annotations)
+				}
+			},
+		},
+		{
+			name: "missing metadata",
+			caFn: func() (*crypto.CA, error) {
+				return newTestCACertificate(pkix.Name{CommonName: "signer-tests"}, int64(1), metav1.Duration{Duration: time.Hour * 24 * 60}, time.Now)
+			},
+			initialConfigMapFn: func() *corev1.ConfigMap {
+				return &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns", Name: "trust-bundle",
+						ResourceVersion: "10",
+					},
+					Data: map[string]string{},
+				}
+			},
+			verifyActions: func(t *testing.T, client *kubefake.Clientset) {
+				actions := client.Actions()
+				if len(actions) != 4 {
+					t.Fatal(spew.Sdump(actions))
+				}
+
+				if !actions[0].Matches("get", "configmaps") {
+					t.Error(actions[0])
+				}
+				if !actions[1].Matches("update", "configmaps") {
+					t.Error(actions[1])
+				}
+				if !actions[2].Matches("get", "configmaps") {
+					t.Error(actions[2])
+				}
+				if !actions[3].Matches("update", "configmaps") {
```

Comment:
the second call is unnecessary, we should refactor the controller to issue only a single API call.

Reply:
in addition it is worth to also examine the payload of the request (updated configmap), please add it to your todo list.